crates/database/src/order_execution.rs

```diff
@@ -1,33 +1,106 @@
 use {
-    crate::{auction::AuctionId, OrderUid},
+    crate::{auction::AuctionId, Address, OrderUid},
     bigdecimal::BigDecimal,
-    sqlx::PgConnection,
+    sqlx::{PgConnection, QueryBuilder},
+    std::collections::HashMap,
 };
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct FeeAsset {
```

Comment:
nit: seems like this type is not specific to fees and could be used for any `Asset` not just `FeeAsset`. It could probably be used in other parts of this crate in the future.

```suggestion
pub struct Asset {
```